src/OpenSage.Game/Logic/Orders/OrderProcessor.cs

```diff
@@ -93,7 +93,25 @@ public void Process(IEnumerable<Order> orders)
 
                             foreach (var unit in player.SelectedUnits)
                             {
-                                unit.ProductionUpdate.QueueProduction(objectDefinition);
+                                // Only units that can produce stuff should produce it
+                                unit?.ProductionUpdate.QueueProduction(objectDefinition);
+                            }
+                        }
+                        break;
+
+                    case OrderType.CancelUnit:
+                        {
+                            var queueIndex = order.Arguments[0].Value.Integer;
+
+                            foreach (var unit in player.SelectedUnits)
+                            {
+                                // Only units that can produce stuff should produce it
+                                var productionJob = unit?.ProductionUpdate.ProductionQueue[queueIndex];
```

Comment:
If `unit` is `null` here, then the following line will crash.

Reply:
I misplaced the ?. It should be after ProductionUpdate. I thought i pushed my correction, sorry...